packages/web-app-admin-settings/src/components/Users/GroupSelect.vue

```diff
@@ -56,18 +55,27 @@ export default defineComponent({
   },
   emits: ['selectedOptionChange'],
   setup(props, { emit }) {
-    const selectedOption = ref(props.selectedGroups)
+    const selectedOptions = ref()
     const onUpdate = (event) => {
-      selectedOption.value = event
-      emit('selectedOptionChange', unref(selectedOption))
+      selectedOptions.value = event
+      emit('selectedOptionChange', unref(selectedOptions))
     }
 
     const currentGroups = computed(() => props.selectedGroups)
-    watch(currentGroups, () => {
-      selectedOption.value = props.selectedGroups
-    })
+    watch(
+      currentGroups,
+      () => {
+        selectedOptions.value = props.selectedGroups
+          .map((g) => ({
+            ...g,
+            readonly: g.groupTypes?.includes('ReadOnly')
+          }))
+          .sort((a: any, b: any) => b.readonly - a.readonly)
```

Comment:
🙈

Reply:
I know, I know 😄 But even when using the correct types here, TS would complain that you can't subtract boolean values from another.